### DISCUSSION TITLE
connect two flexible plates with bolts?

### DISCUSSION THREAD

**Discussion by birdshao** [jgerstmayr/EXUDYN#22] [Q&A on usage of Exudyn] [answered]
How to describe the connection between two flexible plates with bolts? which connector should I use?  Two plates are described through ObjectFFRFreducedOrderInterface and the nodes on the connect face are chosen as two group markers. I connect two group markers by CartesianSpringDamper. Without any input, two plates will move with each other because the inital length of spring is set to be zero but in fact the two group markers are not at the same positon. The relative position between two group markers will cause initial forces which drives the plate. And the initial length of spring is hard to get since the nodes are mesh by ngsolve.

**Comment by jgerstmayr** [comment, marked as answer]
Connnecting two plates with bolts may require to use contact. This is more involved to realize.
If you would like to connect parts of the plates with each other, the approach of the CartesianSpringDamper seems to work well (and efficient). The initial length can be read out in two ways:
1) you measure the midpoint of the marker nodes using the FEM function GetNodePositionsMean(...), see Examples/CMSexampleCourse.py.
With this information, you can define an offset for MarkerSuperElementRigid; in this way, you can shift your Marker position to the point at the interface of the two plates
2) you first assemble the model (otherwise compute functions do not work!) and then run for your objectNumber of your CartesianSpringDamper:
x = mbs.GetObjectOutput(objectNumber, exu.OutputVariableType.DisplacementLocal) 
==> this gives you the (initial = current) offset x of the two marker positions
see again Examples/CMSexampleCourse.py

Note that Examples/CMSexampleCourse.py currently includes both offset shifts, one should in fact be deactivated to run properly. This will be corrected in the next update.
